shell/platform/android/io/flutter/embedding/engine/dynamicfeatures/PlayStoreDynamicFeatureManager.java

```diff
@@ -0,0 +1,336 @@
+              sessionIdToName.remove(state.sessionId());
+              break;
+            }
+          case SplitInstallSessionStatus.CANCELING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install canceling.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.PENDING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install pending.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.REQUIRES_USER_CONFIRMATION:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) install requires user confirmation.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.DOWNLOADING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) downloading.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.DOWNLOADED:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) downloaded.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          case SplitInstallSessionStatus.INSTALLING:
+            {
+              Log.d(
+                  TAG,
+                  String.format(
+                      "Module \"%s\" (sessionId %d) installing.",
+                      sessionIdToName.get(state.sessionId()), state.sessionId()));
+              break;
+            }
+          default:
+            Log.d(TAG, "Status: " + state.status());
+        }
+      }
+    }
+  }
+
+  public PlayStoreDynamicFeatureManager(@NonNull Context context, @Nullable FlutterJNI flutterJNI) {
+    this.context = context;
+    this.flutterJNI = flutterJNI;
+    splitInstallManager = SplitInstallManagerFactory.create(context);
+    listener = new FeatureInstallStateUpdatedListener();
+    splitInstallManager.registerListener(listener);
+    sessionIdToName = new HashMap();
+    sessionIdToLoadingUnitId = new HashMap();
+  }
+
+  public void setJNI(@NonNull FlutterJNI flutterJNI) {
+    this.flutterJNI = flutterJNI;
+  }
+
+  private boolean verifyJNI() {
+    if (flutterJNI == null) {
+      Log.e(TAG, "No FlutterJNI provided.");
```

Comment:
I think this appears without context. Be a bit more detailed. You must set this to integrate split modules before calling dart apis or loading assets via platform channel.